Source/WebKit/NetworkProcess/Downloads/DownloadManager.cpp

```diff
@@ -127,7 +127,10 @@ void DownloadManager::cancelDownload(DownloadID downloadID, CompletionHandler<vo
         pendingDownload->cancel(WTFMove(completionHandler));
         return;
     }
+    // There is a chance that the DownloadManager has called downloadFinished() and the download with downloadID has been removed frome m_downloads.
+#if !PLATFORM(GTK) && !PLATFORM(WPE)
     ASSERT_NOT_REACHED();
+# endif
```

Comment:
Please don't make the behavior platform-specific! You can just remove the assert.